src/subscription_manager/gui/registergui.py

```diff
@@ -556,6 +602,16 @@ def _on_screen_register_finished(self, obj):
         It may need to auto attach, etc. The 'finished' signal indicates register
         and attach are finished, while 'register-finished' is just the first part."""
 
+        # if self.info.get_property('server-info-changed'):
```

Comment:
I think the current 'RegisterWidget._on_connection_info_changed' should be replaced with 'RegisterWidget._on_server_info_changed'. Changes to RegisterInfo's 'server-info' will trigger that callback
(if the current handlers for serverInfo's 'notify::hostname' signals (and friends) are updated for server-info).

Always persisting the config post register seems correct to me. 

But, I think this may make more sense in a handler for RegisterWidget's 'register-finished' instead of it's handler for each screens 'screen-register-finished'. Ie, I think this should move to RegisterWidget.do_register_finished()
 
(The Screens emit a 'screen-register-finished' that is handled by RegisterWidget._on_screen_register_finished, which then makes RegisterWidget emit it's 'register-finished'. The RegisterWidget.do_register_finished will then be
the first handle called (even before a handler setup by a parent dialog like RegisterDialog). Given config doesn't seem like something that a different parent dialog should need to have to change, RegisterWidget.do_register_finished() seems reasonable).